File: packages/oauth/oauth-client-expo/ios/Records.swift
import ExpoModulesCore

struct EncodedJWK: Record {
  @Field
  var kty: String

  @Field
  var crv: String

  @Field
  var kid: String

  @Field
  var x: String

  @Field
  var y: String

  @Field
  var d: String

  @Field
  var alg: String
}

struct VerifyOptions: Record {
  @Field
  var audience: String?

  @Field
  var clockTolerance: Double?

  @Field
  var issuer: String?

  @Field
  var maxTokenAge: Double?

  @Field
  var subject: String?

  @Field
  var typ: String?

  @Field
  var currentDate: Date?

  @Field
  var requiredClaims: [String]?
}

struct VerifyResult: Record {
  @Field
  var payload: String

  @Field
  var protectedHeader: [String: Any]
}

File: packages/oauth/oauth-client-expo/android/src/main/java/expo/modules/atprotooauthclient/Records.kt
package expo.modules.atprotooauthclient

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class EncodedJWK : Record {
  @Field
  var kty: String = ""

  @Field
  var crv: String = ""

  @Field
  var kid: String = ""

  @Field
  var x: String = ""

  @Field
  var y: String = ""

  @Field
  var d: String = ""

  @Field
  var alg: String = ""
}

class VerifyOptions : Record {
  @Field
  var audience: String? = null

  @Field
  var checkTolerance: Double? = null

  @Field
  var issuer: String? = null

  @Field
  var maxTokenAge: Double? = null

  @Field
  var subject: String? = null

  @Field
  var typ: String? = null

  @Field
  var currentDate: Double? = null

  @Field
  var requiredClaims: Array<String>? = null
}

class VerifyResult : Record {
  @Field
  var payload: String = ""

  @Field
  var protectedHeader: Map<String, Any> = emptyMap()
}
